packages/components/src/components/Text/Text.tsx

```diff
@@ -0,0 +1,77 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { type HTMLAttributes, type DetailedHTMLProps, type ReactNode, forwardRef, type ForwardedRef } from 'react';
+
+import { type CommonAriaProps } from '#/helpers/aria';
+import { cn } from '#/utilities/styleUtilities';
+
+import styles from './styles/text.module.scss';
+
+type TextWeight = 'default' | 'strong' | 'x-strong';
+type TextSize = 'x-small' | 'small' | 'medium' | 'large';
+type TextColor = 'default' | 'weak' | 'x-weak' | 'disabled' | 'negative' | 'positive' | 'warning' | 'interactive';
+type BoxColor = 'neutral' | 'selected' | 'disabled' | 'positive' | 'negative' | 'warning';
+
+type TagType = 'a' | 'abbr' | 'address' | 'em' | 'label' | 'li' | 'span' | 'strong' | 'time' | 'p';
+
+export type TextProps<TTag extends TagType = 'span'> = CommonAriaProps &
+    DetailedHTMLProps<HTMLAttributes<TextElementType<TTag>>, TextElementType<TTag>> & {
+        size?: TextSize;
+        weight?: TextWeight;
+        as?: TTag;
+        color?: TextColor;
+        /** @description optional color prop that uses the inverse box color when accessibility contrast is needed */
+        boxColor?: BoxColor;
+        children?: ReactNode;
```

Comment:
Have a look at all other components, they have some kind of docs in the type, please add it too (same in heading)